Source/UltraleapTrackingCore/Private/LeapUtility.cpp

```diff
@@ -184,4 +193,46 @@ void FLeapUtility::ConvertFStringArrayToCharArray(const TArray<FString>& FString
 void FLeapUtility::SetLastArrayElemNull(const char*** ConstCharArrayPtr, int32 LastIdx)
 {
 	(*ConstCharArrayPtr)[LastIdx] = NULL;
-}
\ No newline at end of file
+}
+
+FString FLeapUtility::GetAnalyticsData(size_t& Size)
+{
+	FAnalytics Analytics;
+	Analytics.telemetry.app_title = FApp::GetProjectName();
+
+#if WITH_EDITOR
+	Analytics.telemetry.app_type = "editor";
+#else
+	Analytics.telemetry.app_type = "build";
+#endif
+
+	Analytics.telemetry.engine_name = "Unreal";
+	FString UnrealVersion = FString::FromInt(ENGINE_MAJOR_VERSION);
+	UnrealVersion += ".";
+	UnrealVersion += FString::FromInt(ENGINE_MINOR_VERSION);
+	Analytics.telemetry.engine_version = UnrealVersion;
+	Analytics.telemetry.installation_source = "github";
+	Analytics.telemetry.plugin_version = FString();
+
+
+	TSharedPtr<IPlugin> Plugin = IPluginManager::Get().FindPlugin(FString("UltraleapTracking"));
+	if (Plugin.IsValid())
+	{
+		const FPluginDescriptor& PluginDescriptor = Plugin->GetDescriptor();
+		Analytics.telemetry.plugin_version = PluginDescriptor.VersionName;
+
+		if (Plugin->GetLoadedFrom() == EPluginLoadedFrom::Engine)
+		{
+			Analytics.telemetry.installation_source = "EpicGamesLauncher";
+		}
+	}
+
+	FString SerializedJson;
+	bool bConverted = FJsonObjectConverter::UStructToFormattedJsonObjectString<TCHAR, TPrettyJsonPrintPolicy>(FAnalytics::StaticStruct(), &Analytics, SerializedJson);
+	if (!bConverted)
+	{
+		UE_LOG(UltraleapTrackingLog, Error, TEXT("FLeapUtility::GetAnalyticsData Failed Json conversion"));
+	}
+	Size = SerializedJson.Len() + 1;
```

Comment:
Why +1? The string terminator?

Reply:
The conversion does not count for the last char witch is null, same done in [OpenXR](https://gitlab.ultrahaptics.com/openxr/ultraleap-openxr/-/blob/main/src/hand-tracking/UltraleapTrackingConnection.cpp#L56:~:text=result%20%3D%20LeapSetConnectionMetadata(connectionHandle%2C%20metadata.c_str()%2C%20metadata.length()%20%2B%201)%3B)